### PR TITLE
add support for PersonPath22 dataset

### PR DESCRIPTION
**Pull request by alebergamo**
Hi Jonathon, this is Alessandro, we got in touch by email (bergamo@amazon.com). This is the pull request for the addition of our new PersonPath22 benchmark. As mentioned in the email, the paper is not public yet, but it would be great if you could take a look at the code and tell me if you want to chance something.

Thanks,
Alessando

### CODE REVIEW

**Comment by JonathonLuiten** [comment]
Hey Alessandro.

I just looked at this now, most of changes you have suggested are independent of everything else here so are safe to merge in as they won't cause anything else to break.

However, this is not the case for your suggested changes to the file: 'trackeval/datasets/mot_challenge_2d_box.py'

Is there someway you could rearrange this code to not have to change the code of this file? If so I would be happy to merge this in.

Furthermore, it would be good to wait until the related paper is public with a link that we can link to before merging this in as well.

Regards,
Jonathon

**Comment by alebergamo** [comment]
Hi Jonathon,

regarding the code changes in mot_challenge_2d_box.py, I am just re-using some already-existing patterns I found in the codebase, so to add the support for crowd-regions in the mot_challenge_2d_box.py format.

More in detail, the crowd region was already defined in the MOT format [here](https://github.com/JonathonLuiten/TrackEval/blob/master/trackeval/datasets/mot_challenge_2d_box.py#L78), but it was not fully implemented. I then implemented the crowd filtering exactly as done for the Kitti dataset. As you can see, my filtering method [here](https://github.com/JonathonLuiten/TrackEval/pull/92/files#diff-bf9a499c48b8d86bb403823f99380c4892b7a948ea2e366b0776d35f66fad7a4R201-R204) is exactly the same as the one for Kitti [here](https://github.com/JonathonLuiten/TrackEval/blob/master/trackeval/datasets/kitti_2d_box.py#L146-L150). Later, the filter is used in the same way: [my_change](https://github.com/JonathonLuiten/TrackEval/pull/92/files#diff-bf9a499c48b8d86bb403823f99380c4892b7a948ea2e366b0776d35f66fad7a4R279-R283) vs [Kitti](https://github.com/JonathonLuiten/TrackEval/blob/master/trackeval/datasets/kitti_2d_box.py#L217-L221) . The final remove of the bboxes at eval time also follows the same pattern [my_change](https://github.com/JonathonLuiten/TrackEval/pull/92/files#diff-bf9a499c48b8d86bb403823f99380c4892b7a948ea2e366b0776d35f66fad7a4R394-R396) vs [Kitti](https://github.com/JonathonLuiten/TrackEval/blob/master/trackeval/datasets/kitti_2d_box.py#L331-L334)

In summary, this change extends the functionalities of mot_challenge_2d_box.py, which could be later be useful for other datasets as well. If you are concern about breaking the eval for MOT17, do you have unit tests or other integration tests I can run?

Having said that, if this is not how you want to proceed, I can just copy and paste the entire file and add a new one for PersonPath22, just let me know.

The paper can be found here: https://www.amazon.science/publications/large-scale-real-world-multi-person-tracking and this is the official ECCV record id for our paper https://eccv2022.ecva.net/program/accepted-papers/

Best,
Alessandro

**Comment by alebergamo** [comment]
I just updated the URL in the README which now points to the dataset webpage (which also has the paper in PDF)

**Comment by alebergamo** [comment]
Hi Jonathon,

just following-up on this, please let me know if you have questions.

Thanks, Alessandro

**Comment by JonathonLuiten** [comment]
Hey!

Breaking the eval for MOT17 is exactly what I am worried about. We don't have trustworthy integration or unit tests at the moment, so I would rather you didn't change these files at all.

You're right that what you did was implement crowd filtering from kitti into motchallenge, but that to be consistent with how motchallenge has always been evaluated these shouldn't be implemented there. 

The easiest thing from my side is if you copy all the code over, and then you're definitely not making any breaking changes, and I can accept this straight away.

Jonathon

**Comment by alebergamo** [comment]
Hi Jonathon,

ok sounds good. I reverted the changes made in mot_challenge_2d_box.py. 

Best, Alessandro
